bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/io/ChannelPool.java

```diff
@@ -22,41 +31,33 @@
 import io.grpc.Channel;
 import io.grpc.ClientCall;
 import io.grpc.ClientInterceptors.CheckedForwardingClientCall;
+import io.grpc.ManagedChannel;
 import io.grpc.Metadata;
 import io.grpc.MethodDescriptor;
 
-import java.io.Closeable;
-import java.io.IOException;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.concurrent.atomic.AtomicBoolean;
-import java.util.concurrent.atomic.AtomicInteger;
-import java.util.concurrent.atomic.AtomicReference;
-import java.util.logging.Logger;
-
 /**
  * Manages a set of ClosableChannels and uses them in a round robin.
  */
-public class ChannelPool extends Channel implements Closeable {
+public class ChannelPool extends ManagedChannel {
 
-  protected static final Logger log = Logger.getLogger(ChannelPool.class.getName());
+  protected static final Logger LOG = new Logger(ChannelPool.class);
 
   public interface ChannelFactory {
-    Channel create() throws IOException;
+    ManagedChannel create() throws IOException;
   }
 
-  private final AtomicReference<ImmutableList<Channel>> channels = new AtomicReference<>();
+  private final AtomicReference<ImmutableList<ManagedChannel>> channels = new AtomicReference<>();
   private final AtomicInteger requestCount = new AtomicInteger();
   private final ImmutableList<HeaderInterceptor> headerInterceptors;
   private final AtomicInteger totalSize;
   private final ChannelFactory factory;
   private final String authority;
-  
-  private final AtomicBoolean closed = new AtomicBoolean(false);
+
+  private boolean shutdown = false;
```

Comment:
It should be shutdown.  All of the underlying channels were shutdown() and should all return true for isShutdown().  I'll change isShutdown() to return this.shutdown.